src/main/java/com/google/edith/QueryItems.java

```diff
@@ -0,0 +1,151 @@
+package com.google.edith;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.common.base.Splitter;
+import com.google.edith.servlets.Item;
+import com.google.edith.servlets.Receipt;
+import com.google.gson.Gson;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds the items on past receipts that have expired by the time of the user's next shopping trip.
+ */
+public class QueryItems {
+
+  private final DatastoreService datastore;
+  private UserService userService;
+
+  public QueryItems(DatastoreService datastore, UserService userService) {
+    this.datastore = datastore;
+    this.userService = userService;
+  }
+
+  public String findExpiredItems() {
+    List<Receipt> receipts = queryReceipts();
+    Set<Item> itemsToBuy = new HashSet<Item>();
+
+    for (Receipt receipt : receipts) {
+      double daysPassedSinceReceipt = findTimePassed(receipt.getDate());
+      Item[] items = receipt.getItems();
+
+      for (Item item : items) {
+        String expiration = item.expiration();
+        if (expiration.equals("NO_EXPIRATION")) {
+          continue;
+        }
+
+        List<String> expirationNumberAndUnit = Splitter.on(" ").splitToList(expiration);
```

Comment:
I think we can move the " " into a private static final str at the top of the file